src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingService.java

```diff
@@ -0,0 +1,363 @@
+import com.faforever.client.notification.PersistentNotification;
+import com.faforever.client.notification.Severity;
+import com.faforever.client.notification.TransientNotification;
+import com.faforever.client.player.Player;
+import com.faforever.client.player.PlayerService;
+import com.faforever.client.preferences.PreferencesService;
+import com.faforever.client.preferences.event.MissingGamePathEvent;
+import com.faforever.client.rankedmatch.MatchmakerInfoMessage;
+import com.faforever.client.remote.FafServerAccessor;
+import com.faforever.client.remote.FafService;
+import com.faforever.client.remote.domain.GameLaunchMessage;
+import com.faforever.client.remote.domain.MatchCancelledMessage;
+import com.faforever.client.remote.domain.MatchFoundMessage;
+import com.faforever.client.remote.domain.MatchmakingState;
+import com.faforever.client.remote.domain.PartyInfoMessage;
+import com.faforever.client.remote.domain.PartyInviteMessage;
+import com.faforever.client.remote.domain.PartyKickedMessage;
+import com.faforever.client.remote.domain.SearchInfoMessage;
+import com.faforever.client.teammatchmaking.MatchmakingQueue.MatchingStatus;
+import com.faforever.client.teammatchmaking.Party.PartyMember;
+import com.faforever.client.util.IdenticonUtil;
+import com.google.common.eventbus.EventBus;
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledFuture;
+
+@Lazy
+@Service
+@Slf4j
+public class TeamMatchmakingService implements InitializingBean {
+
+  private final FafServerAccessor fafServerAccessor;
+  private final PlayerService playerService;
+  private final NotificationService notificationService;
+  private final PreferencesService preferencesService;
+  private final FafService fafService;
+  private final EventBus eventBus;
+  private final I18n i18n;
+  private final TaskScheduler taskScheduler;
+  private final GameService gameService;
+
+  @Getter
+  private final Party party;
+  @Getter
+  private final ObservableList<MatchmakingQueue> matchmakingQueues = FXCollections.observableArrayList();
+  private final List<ScheduledFuture<?>> leaveQueueTimeouts = new LinkedList<>();
+
+  private volatile boolean matchFoundAndWaitingForGameLaunch = false;
+  private boolean queuesAdded = false;
+
+  @Override
+  public void afterPropertiesSet() throws Exception {
```

Comment:
If not needed remove Initializing bean or use `@RequiredArgsConstructor` and do all the current constructor logic here.